docs/zkapps/o1js/bitwise-operations.mdx

```diff
@@ -0,0 +1,456 @@
+---
+title: Bitwise Operations
+hide_title: true
+description: Bitwise operations in o1js are implemented as gadgets.
+keywords:
+  - gadgets
+  - bitwise operations
+  - o1js
+---
+
+:::info
+
+Please note that zkApp programmability is not yet available on Mina Mainnet, but zkApps can now be deployed to Berkeley Testnet.
+
+:::
+
+# Bitwise Operations
+
+In o1js, bitwise operations are implemented as [gadgets](/zkapps/o1js/gadgets).
+
+- [and()](#and)
+- [not()](#not)
+- [xor()](#xor)
+- [addMod32()](#addmod32)
+- [divMod32()](#divmod32)
+- [leftShift32()](#leftshift32)
+- [leftShift64()](#leftshift64)
+- [rightShift32()](#rightshift32)
+- [rightShift64()](#rightshift64)
+- [rotate32()](#rotate32)
+- [rotate64()](#rotate64)
+- [rangeCheck32()](#rangecheck32)
```

Comment:
FYI - some of these aren't in the current release as planned, but in this PR: https://github.com/o1-labs/o1js/pull/1259
(but maybe we can merge the PR before release, cc @Trivo25)

Reply:
The PR is ready from my side, except two or three tiny follow ups - I am just waiting on the crypto stamp, unless you think thats not needed